File: Front/TS/src/domain/bets/IBetRepository.ts
import { Bet } from "./Bet";
import { BetSummary } from "./BetSummary";

export interface IBetRepository {
    GetProofAsync(BetId: string): Promise<string>;
    CompleteAsync(betId: string, isSuccess: boolean, proof?: string): Promise<void>;
    AnswerAsync(BetId: string, Answer: boolean): Promise<void>;
    GetAllAsync(): Promise<BetSummary[]>;
    Save(bet: Bet) : Promise<void>;
}

File: Front/.Net Maui/BetFriends.Domain/Bets/IBetRepository.cs
using BetFriends.Domain.Features.CreateBet;
using BetFriends.Domain.Features.RetrieveBets;

namespace BetFriends.Domain.Bets;

public interface IBetRepository
{
    Task AnswerBetAsync(string betId, bool answer);
    Task<IEnumerable<RetrieveBetsItemResponse>> RetrieveBetsAsync();
    Task CompleteBetAsync(string betId, bool isSuccess, string? proof, CancellationToken cancellationToken);
    Task<byte[]> RetrieveProofAsync(string betId);
    Task SaveAsync(Bet bet, CancellationToken cancellationToken);
}
